### ISSUE TITLE
Everything is out there... people just don't find it before they stop looking...

### ISSUE THREAD

**Issue by fjug** [imagej/tutorials#20]
And yes... they should just look longer and be more persistent... but let me tell you... they aren't and this is how it is.

If you still want these people develop for you (or only just use things you did), I suggest the following (which we just discussed in the ImageJ hackathon dungeon on Sept. 6th).

Come up with some typical user profiles and assemble pages tailored to suit the needs of exactly these people and their level of knowledge. The most important thing here is not adding new information to the wiki etc., but making this content easily digestible and available in a linear sequence of things to go through -- a thing that is VERY hard for people entering this infinite world of stuff you should know to not feel the pain of starting.

In some sense such a page is like a thing a person would want... say... _How can I write an ImageJ2 Plugin_? And lets say we now discuss this not in terms of a user that is a geek already but a computer affine biologist. Obviously this will take a while to learn then because you also need to learn about eclipse, maven, ImgLib2, etc., so maybe the page starts with a TOC that shows a good sequence to learn about this things + an time one would expect for this. The person finding such a page is then prepared to have to learn shitloads of things and most likely have to spend like 5 complete days. This is how it is, but hey... now at least the hard facts are clear from moment one and not only visible after several hours of painful random search for a solution the searching person has NO CLUE ABOUT.

Note that the kind of pages I speak about are at the end only link to other existing pages, the problem of creating them is therefore not of any technical difficulty but really 'only' about being smart about how to guide people through the 'land of pain' without making them falter and stop the journey before reaching their goal.

Every time we see a person that start on an endeavor to 'the land of pain' we might help this person to easily assemble a wiki page where he simultaneously types in his experience... in this way we at least get a trace of his trip -- most likely a good starting point to then make a page of the kind I have in mind.

Hope my thoughts make any sense in the way I describe them,
Florian

PS: any page of this kind in even the rawest of shapes I would be happy read through and help improving... let me know!

**Comment by dietzc** [comment]
https://github.com/biobv2015/knip-developer-guide this is what I provide to my students... it helps them a lot.

**Comment by hinerm** [comment]
thanks for this @fjug

**Comment by ctrueden** [comment]
Great, totally agreed. Now: can someone who is not me or @hinerm start this
effort at this very hackathon, within the next couple of days????

Of course, I also pledge to help improve and further organize it!

**Comment by etarena** [comment]
i'm on it.  (providing my newbie-perspective perhaps will be helpful...)

**Comment by etarena** [comment]
just an aside...  I think a lot of the problems with mis-understanding come from two places:
1) terminology (using more 'simple' terms to describe things for those who are not deep within the cs world - i’m not saying ‘dumb it down’ but ‘clear it up’)
2) missing context (what is the point of this tool?  why are we using it?  where does it fit in in the big scheme of things?)

**Comment by ctrueden** [comment]
So, the central starting point for learning ImageJ is now intended to be http://imagej.net/Learn. And for developers: http://imagej.net/Develop.

We are working on unifying all the developer tutorials as Beaker notebooks in this repository. Several complementary resources will be (at least largely) unified here, such as `scifio/scifio-tutorials` and `imagej/imagej-scripting`.

I am closing this issue, because I do not know what else to do to achieve your vision, other than to make the above links as good as possible. But if you feel there is still something central which is fundamentally missing, please reopen and describe your concrete idea for what we should do.
